### ISSUE TITLE
Add label "app" to PodMetadata doesn't work

### ISSUE THREAD

**Issue by Danipiario** [prometheus-operator/prometheus-operator#3706]
**What happened?**
Label "app" added with PodMetadata is not propagated to prometheus pod

**Did you expect to see something different?**
With this Prometheus configuration:
```
apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
...
spec:
...
  podMetadata:
    labels:
      app: custom-label
...
```
I expect to have a Prometheus pod like this:
```
kind: Pod
apiVersion: v1
metadata:
...
  labels:
    app: custom-label
...
spec:
...
```
but the Prometheus pod is always created with 
```
kind: Pod
apiVersion: v1
metadata:
...
  labels:
    app: prometheus
...
spec:
...
```

**How to reproduce it (as minimally and precisely as possible)**:

**Environment**

* Prometheus Operator version:

    `prometheus-operator:0.37.0`
    

* Kubernetes version information:

    `Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2-0-g52c56ce", GitCommit:"b66f2d3", GitTreeState:"clean", BuildDate:"2020-08-11T15:2
3:53Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.3+2fbd7c7", GitCommit:"2fbd7c7", GitTreeState:"clean", BuildDate:"2020-10-15T16:25:
45Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
`
    

* Kubernetes cluster kind:

    Openshift cluster 4.5.16

* Manifests:
* Prometheus Operator Logs:

**Anything else we need to know?**:

**Comment by simonpasquier** [comment]
The `app` and `prometheus` labels are reserved by the operator to configure the `prometheus-operated` service IIUC. We need either to document this limitation in the documentation or use a different label names (prefixed by "operator.prometheus.io"?). I don't have enough background to decide which option is appropriate.
cc @s-urbaniak @brancz

**Comment by brancz** [comment]
That is correct @simonpasquier. If we change these, then we should only change them once and clean them up once and for all. If we do that, then I think we should change these to be the official Kubernetes app labels.

**Comment by bygui86** [comment]
Hi guys,

I'm using the `kube-prometheus release-0.4` with operator `0.35.1` and no matter which label I add, they are completely ignored by the operator.

For example:
```yaml
apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
  name: test
spec:
  baseImage: quay.io/prometheus/prometheus
  version: v2.22.1
  replicas: 1
  logFormat: json
  logLevel: info
  resources:
    requests:
      cpu: "3"
      memory: 4G
    limits:
      cpu: "6"
      memory: 6G
  retention: 60d
  retentionSize: 500GB
  serviceAccountName: prometheus
  securityContext:
    runAsUser: 1000
    fsGroup: 2000

  # HERE
  podMetadata:
    labels:
      domain: my-domain
      feature: my-cool-monitoring-feature

  alerting:
    alertmanagers:
      - name: alertmanager-main
        namespace: monitoring
        port: web
  ruleNamespaceSelector: {}
  ruleSelector:
    matchLabels:
      app: prometheus
      role: alert-rules
  podMonitorNamespaceSelector: {}
  podMonitorSelector: {}
  serviceMonitorNamespaceSelector: {}
  serviceMonitorSelector: {}
  storage:
    volumeClaimTemplate:
      metadata:
        name: prometheus
        
        # AND HERE
        labels:
          domain: my-domain
          feature: my-cool-monitoring-feature
          type: ssd

      spec:
        accessModes:
          - ReadWriteOnce
        resources:
          requests:
            storage: 750Gi
  nodeSelector:
    kubernetes.io/os: linux
```

Is the behaviouor still correct or does it look weird?

Thanks for your help!

**Comment by simonpasquier** [comment]
@bygui86 

> Hi guys,

:wave: there aren't only "guys" in this community :smirk: 

`0.35.1` is very old, can you update the version of the operator?

**Comment by bygui86** [comment]
ok sorry! let me rephrase!

hi everybody!!

@simonpasquier I'm running kubernetes on `GKE` with version `v1.17.13-gke.600`, so looking at the compatibility matrix, more recent versions should not work. If you think like `release-0.5` could work, I can give it a try!

**Comment by simonpasquier** [comment]
@bygui86 you should be able to use prometheus-operator v0.43.2 with GKE v1.17. I don't think you need to update the kube-prometheus version.

**Comment by bygui86** [comment]
@simonpasquier I tried already to do some upgrades on prometheus-operator: from v0.35.1 first to 0.36.0, then 0.37.0, then 0.38.0 but they require also CRDs upgrade, so I gave up thinking that the entire kube-prometheus release was required.

I try to upgrade both CRDS and prometheus-operator to the latest versions and let you know!

**Comment by bygui86** [comment]
@simonpasquier thanks for your tip! I did an upgrade of all components in the prometheus environment and works perfectly.
Basically I can say that kube-prometheus `release-0.6` and `master` work also with Kubernetes `1.17.x`.

**Comment by hooten** [comment]
I would like to work to update all of the `app` label names to use the recently added `app.kubernetes.io/name` label name. I need this for my use case anyway.
